### ISSUE TITLE
Question: using `self` instead of defining CIDR on security group

### ISSUE THREAD

**Issue by nathanielks** [segmentio/stack#76]
Hiya, team!

Reading over your security group module and was curious: why define a CIDR for the internal security groups when you could set `self = true`? I believe they'll accomplish the same thing and it'd be one less variable to pass around!

So instead of:

```hcl
resource "aws_security_group" "internal_elb" {
  // trimmed...
  ingress {
    from_port   = 80
    to_port     = 80
    protocol    = "tcp"
    cidr_blocks = ["${var.cidr}"]
  }
}
```

Do:

```hcl
resource "aws_security_group" "internal_elb" {
  // trimmed...
  ingress {
    from_port   = 80
    to_port     = 80
    protocol    = "tcp"
    self        = true
  }
}
```

From Terraform's docs:
> If true, the security group itself will be added as a source to this ingress|egress rule.

**Comment by achille-roussel** [comment]
Could you submit a PR for this?

**Comment by nathanielks** [comment]
@achille-roussel I sure can. Circling back to this, I also realize this may not be what we want as the load balancers would also need to have this security group attached for `self = true` to work.

TL;DR: I think it's good as is.

**Comment by achille-roussel** [comment]
Alright, your call 👍
